src/app/components/seedPhraseInput/index.tsx

```diff
@@ -167,6 +175,16 @@ export default function SeedPhraseInput({
     });
   };
 
+  const handlePaste = (pastedText: string) => {
+    const splitPastedText = pastedText.split(' ');
+    splitPastedText.forEach((text, index) => {
+      setSeedInputValues((prevSeed) => {
+        prevSeed[index] = text;
+        return [...prevSeed];
+      });
+    });
```

Comment:
not sure if react optimises this for you, but does it result in 12 or 24 renders? you could probably just setState once

Reply:
@jordankzf to follow up